tests/console/validate_lvm.pm

```diff
@@ -39,9 +40,39 @@ sub run {
         assert_equals($vol_status, 'ACTIVE', "Volume is Inactive");
     }
 
+    # Sum up the Physical Extents across all Physical Volumes within each Volume group
+    # of a given system. So it gives us : $PEs_in_VGs{$vg} = [@extents_in_vg]
+    # Then do the same for the free extents, so we can compare total PEs with total logical extents.
     record_info('equal extents', 'Verify sum of logical extents corresponds to physical extent size');
-    my $pvTotalPE = script_output q[pvdisplay|grep "Total PE" | awk '{print $3}'];
-    my $pvFreePE  = script_output q[pvdisplay|grep "Free PE" | awk '{print $3}'];
+    my @PV = split(/\n/, script_output qq[pvdisplay |grep "PV Name" |awk '{print \$3}']);
```

Comment:
the idea behind the test_data is to avoid this kind of code where we rely on previous result from previous command (which could be wrong) which at the end make the code easy to understand. We should try to set this list `@PV` and `@VG` as test data.
Another complementary approach that has been used and make test easy to read is to run the script_output command at the beginning outside the loop, get what you need using regex with named capture groups and then do some comparison with our Assert library for Perl.

Reply:
I agree. I did not modify this module to use test data, as I guess its purpose is to validate any lvm setup. Then indeed, using capture groups would be a lot faster.